### ISSUE TITLE
Prompt for save before continue

### ISSUE THREAD

**Issue by nickdrozd** [jacktasia/dumb-jump#201]
When I use `dumb-jump-go-prompt`, the following sequence of events
transpires (as far as I can tell, I could be wrong of course):

1. It prompts `Jump to:` and then I enter `some-function`.

2. It searches the project for `some-function`. 

3. If it finds multiple definitions, it displays a popup menu for me
   to specify. I pick the definition from `file.blah`
   
4. It checks whether `file.blah` has been modified.
   
5. If the file has been modified, it prompts `file.blah has been
   modified so we may have the wrong location. Continue?` An answer of
   `y` will jump to the location that was found in 2, which may be
   inaccurate. An answer of `n` bails on the whole search.
   
In other words, the choice presented in step 5 is between getting a
result that might be wrong and getting nothing. I think it would be
nicer instead to get a prompt at step 5 saying something like
`file.blah has been modified, save it? (y or n)`. An answer of `n`
would carry on using the possibly-inaccurate result from step 2, while
an answer of `y` would save `file.blah` and then re-search just that
file (not the whole project, as in step 2) for `some-function` and
jump to the correct location.

**Comment by jacktasia** [comment]
Thanks for opening this issue! I agree that it makes sense to add something like this. I am thinking of adding something where it would give you more options like `[y]es, continue`, `[s]ave and continue`, `[n]o, stop` this way would allow backward compatible muscle memory :slightly_smiling_face: and it would allow other future options or even an optional hook where people could add their own via customization (letter -> function).

Let me think about this a bit more but I'll try to PR something for this soon. Thanks again!
